nntrainer/layers/layer_context.cpp

```diff
@@ -326,6 +326,25 @@ Tensor &RunLayerContext::getOutgoingDerivative(unsigned int idx) {
   return getInputGrad(idx);
 }
 
+bool RunLayerContext::validateDerivatives() {
+  auto num_in = getNumInputs();
+  auto num_out = getNumOutputs();
+
+  for (unsigned int i = 0; i < num_in; ++i) {
```

Comment:
If we are going to check this validationDerivatives in every layer, then we do not need to check input and output both.